### PR TITLE
FEAT: groupby rolling

### LINKED ISSUE #364: FEAT: implement `groupby.rolling`
### Is your feature request related to a problem? Please describe
`groupby.rolling` is a frequet operation in financial scenarios. Alghouth there is an [alternative way](https://gist.github.com/UranusSeven/ff6ef4c43e60957d76dbdb6e6125ecdd) to do the same thing, we'd better support it.

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?
Implement groupby rolling.

## Related issue number

Resolves #364 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/dataframe/groupby/rolling.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+)
+from ..operands import DataFrameOperand, DataFrameOperandMixin
+from ..utils import parse_index, validate_axis
+
+_PAIRWISE_AGG = ["corr", "cov"]
+
+
+class GroupByRolling(Serializable):
+    _input = KeyField("input")
+    _window = AnyField("window")
```
Comment (qianduoduo0904) [inline python/xorbits/_mars/dataframe/groupby/rolling.py:41]:
```suggestion
    window = AnyField("window")
```

Recommend to use public directly, and the @property is not needed.

File: python/xorbits/_mars/dataframe/groupby/rolling.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    input = KeyField("input")
+    window = AnyField("window")
+    min_periods = Int64Field("min_periods")
+    center = BoolField("center")
+    win_type = StringField("win_type")
+    on = StringField("on")
+    axis = Int32Field("axis")
+    closed = StringField("closed")
+
+    def __init__(
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/groupby/rolling.py:49]: `__init__` is not needed.

File: python/xorbits/_mars/dataframe/groupby/rolling.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    center = BoolField("center")
+    win_type = StringField("win_type")
+    on = StringField("on")
+    axis = Int32Field("axis")
+    closed = StringField("closed")
+    func = AnyField("func")
+    func_args = TupleField("func_args")
+    func_kwargs = DictField("func_kwargs")
+
+    def __init__(
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/groupby/rolling.py:160]: Ditto.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**